tasks/orca_catalog_reporting/orca_catalog_reporting.py

```diff
@@ -129,8 +129,8 @@ def get_catalog_sql() -> text:
             WHERE 
                 (:granule_id is null or cumulus_granule_id=ANY(:granule_id)) and 
                 (:collection_id is null or collection_id=ANY(:collection_id)) and 
-                (:start_timestamp is null or EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC') * 1000>=:start_timestamp) and 
-                (:end_timestamp is null or EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC') * 1000<:end_timestamp)
+                (:start_timestamp is null or EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC')::bigint * 1000>=:start_timestamp) and 
```

Comment:
You will want to do the cast after taking it times 1000 to avoid rounding errors so

`(:start_timestamp is null or EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC')::bigint * 1000` will become `(:start_timestamp is null or (EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC') * 1000)::bigint`

The reason being, `EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC')` may return something like `1638997146.5436` if you cast before taking it `*1000` the casting will round this to `1638997147` which will result in a value of `1638997147000` instead of a value of `16389971465436` which is what you want. This is done on lines 132 and 133.

Reply:
Got complacent with expected precision. Good catch. Fixed.